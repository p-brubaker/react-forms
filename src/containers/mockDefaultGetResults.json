{
    "status": "ok",
    "totalResults": 326,
    "articles": [
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Kristina Panos",
            "title": "Spooky Winners of the Halloween Hackfest Contest",
            "description": "It was a tight race, but the results of the Halloween Hackfest contest are in. We asked you to scare up a terrifying build in one of three categories, and as usual, you didn’t disappoint! The…",
            "url": "https://hackaday.com/2021/11/04/spooky-winners-of-the-halloween-hackfest-contest/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/11/halloweeners-800.jpeg",
            "publishedAt": "2021-11-04T16:01:57Z",
            "content": "It was a tight race, but the results of the Halloween Hackfest contest are in. We asked you to scare up a terrifying build in one of three categories, and as usual, you didn’t disappoint! These three… [+2879 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Adam Fabio",
            "title": "Curious Marc Takes on Sewing Machine Repair",
            "description": "Even the most talented engineers can be stymied by simple repair projects. In this case, repairing a broken sewing machine has [CuriousMarc] all tangled up.  [Marc] is probably best known …read more",
            "url": "https://hackaday.com/2021/11/04/curious-marc-takes-on-sewing-machine-repair/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/10/sew-1-feat.png",
            "publishedAt": "2021-11-04T15:00:00Z",
            "content": "Even the most talented engineers can be stymied by simple repair projects. In this case, repairing a broken sewing machine has [CuriousMarc] all tangled up.  [Marc] is probably best known as a part o… [+1697 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Chris Wilkinson",
            "title": "Weird Al’s Monster Battlestation Is Now Just A Reasonably Fast PC",
            "description": "Wanna be hackers? Code crackers? Slackers. If the vintage computing community ever choose an official anthem, count my vote for It’s All About The Pentiums by “Weird Al” Yankovic. More than twenty …",
            "url": "https://hackaday.com/2021/11/04/weird-als-monster-battlestation-is-now-just-a-reasonably-fast-pc/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/11/WeirdAl.jpg",
            "publishedAt": "2021-11-04T14:00:14Z",
            "content": "Wanna be hackers? Code crackers? Slackers. If the vintage computing community ever choose an official anthem, count my vote for Its All About The Pentiums by Weird Al Yankovic. More than twenty years… [+9263 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Ryan Flowers",
            "title": "Mach 20 In A Wind Tunnel: Yes, It’s Rocket Science",
            "description": "Hypersonic speeds are defined by those exceeding Mach 5, and lately there’s been a lot of buzz about unmanned hypersonic vehicles making test flights. Vehicles returning from orbital flight also …read more",
            "url": "https://hackaday.com/2021/11/04/mach-20-in-a-wind-tunnel-yes-its-rocket-science/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/10/windtunnel-featured.jpg",
            "publishedAt": "2021-11-04T11:00:00Z",
            "content": "Hypersonic speeds are defined by those exceeding Mach 5, and lately there’s been a lot of buzz about unmanned hypersonic vehicles making test flights. Vehicles returning from orbital flight also trav… [+1430 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Kristina Panos",
            "title": "Focus Flower Motivates By Squirting Water In Your Face",
            "description": "When you need to get some tasks done and are short on attention, it’s hard to beat a timer. But whenever you do, it feels pretty darn good. The problem is that when you don’t finish in …",
            "url": "https://hackaday.com/2021/11/04/focus-flower-motivates-by-squirting-water-in-your-face/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/10/focus-flower-bv.jpeg",
            "publishedAt": "2021-11-04T08:00:00Z",
            "content": "When you need to get some tasks done and are short on attention, it’s hard to beat a timer. But whenever you do, it feels pretty darn good. The problem is that when you don’t finish in time, what’s t… [+1628 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Bryan Cockfield",
            "title": "Vintage Computers With a Real Turbo",
            "description": "In prior centuries, it was common practice to tie the operation of a program to a computer’s clock speed. As computers got faster and faster, the programs tied to that slower clock speed some…",
            "url": "https://hackaday.com/2021/11/03/vintage-computers-with-a-real-turbo/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/11/fpga-cpu-apple-featured.png",
            "publishedAt": "2021-11-04T05:00:23Z",
            "content": "In prior centuries, it was common practice to tie the operation of a program to a computer’s clock speed. As computers got faster and faster, the programs tied to that slower clock speed sometimes ha… [+1784 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Bryan Cockfield",
            "title": "More Software-Defined Radio Projects Using DragonOS",
            "description": "DragonOS, a Debian-based Linux distribution specifically packaged for software-defined radio functionality, roared onto the wavelengths during the beginnings of the various pandemic lockdowns last year. Since then [Aaron], the creator …read more",
            "url": "https://hackaday.com/2021/11/03/more-software-defined-radio-projects-using-dragonos/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/10/dragonos-spectrum-main.jpg",
            "publishedAt": "2021-11-04T02:00:37Z",
            "content": "DragonOS, a Debian-based Linux distribution specifically packaged for software-defined radio functionality, roared onto the wavelengths during the beginnings of the various pandemic lockdowns last ye… [+1413 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Lewin Day",
            "title": "Hacking the Mekamon Robot To Add New Capabilities",
            "description": "The Mekamon from Reach Robotics is a neat thing, a robot controlled by a phone app that walks on four legs. [Wes Freeman] decided to hack the platform, giving it a sensor package and enabling some …",
            "url": "https://hackaday.com/2021/11/03/hacking-the-mekamon-robot-to-add-new-capabilities/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/10/7171351529666744137-1.jpg",
            "publishedAt": "2021-11-03T23:00:01Z",
            "content": "The Mekamon from Reach Robotics is a neat thing, a robot controlled by a phone app that walks on four legs. [Wes Freeman] decided to hack the platform, giving it a sensor package and enabling some ba… [+902 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Tom Nardi",
            "title": "Building a Heavy-Duty Log Splitter, One Piece at a Time",
            "description": "With temperatures dropping in the Northern Hemisphere, this is the time of year when many people start processing firewood for the coming winter months. For the city folks, that means chopping a tr…",
            "url": "https://hackaday.com/2021/11/03/building-a-heavy-duty-log-splitter-one-piece-at-a-time/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/10/wfslog_feat.jpg",
            "publishedAt": "2021-11-03T20:00:55Z",
            "content": "With temperatures dropping in the Northern Hemisphere, this is the time of year when many people start processing firewood for the coming winter months. For the city folks, that means chopping a tree… [+1629 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Adam Zeloof",
            "title": "Pringles Can Fluidized Bed Powder Coater Paints Parts Inside And Out",
            "description": "Powder coating is a wonderful way to apply a smooth, colored surface to a part, whether for aesthetic reasons or corrosion protection. Traditionally, powder is applied via a air gun that sprays it …",
            "url": "https://hackaday.com/2021/11/03/pringles-can-fluidized-bed-powder-coater-paints-parts-inside-and-out/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/11/pringles-powder-featured.png",
            "publishedAt": "2021-11-03T18:30:00Z",
            "content": "Powder coating is a wonderful way to apply a smooth, colored surface to a part, whether for aesthetic reasons or corrosion protection. Traditionally, powder is applied via a air gun that sprays it to… [+1995 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Michael Shaub",
            "title": "Keep Calm and Hack On: The Philosophy of Calm Technology",
            "description": "So much smart-tech is really kind of dumb. Gadgets intended to simplify our lives turn out to complicate them. It often takes too many “clicks” to accomplish simple tasks, and …read more",
            "url": "https://hackaday.com/2021/11/03/keep-calm-and-hack-on-the-philosophy-of-calm-technology/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/11/Calm.jpg",
            "publishedAt": "2021-11-03T17:01:23Z",
            "content": "So much smart-tech is really kind of dumb. Gadgets intended to simplify our lives turn out to complicate them. It often takes too many “clicks” to accomplish simple tasks, and they end up demanding o… [+6177 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Tom Nardi",
            "title": "Here Are the Ten Finalists For the Hackaday Prize Wildcard Challenge",
            "description": "Each phase of the 2021 Hackaday Prize challenged designers to reimagine traditional solutions within various fields, from robotics to assistive devices. But for the Reactivate Wildcard, the fifth and final …read more",
            "url": "https://hackaday.com/2021/11/03/here-are-the-ten-finalists-for-the-hackaday-prize-wildcard-challenge/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/11/wildcard21_feat.jpg",
            "publishedAt": "2021-11-03T16:01:24Z",
            "content": "Each phase of the 2021 Hackaday Prize challenged designers to reimagine traditional solutions within various fields, from robotics to assistive devices. But for the Reactivate Wildcard, the fifth and… [+2480 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Dave Rowntree",
            "title": "Simplify 3D Printer Wiring With CAN Bus",
            "description": "[mark] had an interesting idea when looking at all the wiring of a typical 3D printer; Use CAN Bus. There are a lot of wires going to the extruder assembly, and with most designs this thing is flyi…",
            "url": "https://hackaday.com/2021/11/03/simplify-3d-printer-wiring-with-can-bus/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/11/PandaCAN-wide.jpg",
            "publishedAt": "2021-11-03T15:30:00Z",
            "content": "[mark] had an interesting idea when looking at all the wiring of a typical 3D printer; Use CAN Bus. There are a lot of wires going to the extruder assembly, and with most designs this thing is flying… [+1292 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Matthew Carlson",
            "title": "Python Ditches the GILs and Comes Ashore",
            "description": "The Python world has been fractured a few times before. The infamous transition from version 2 to version 3 still affects people today, and there could be a new schism in the future. [Sam Gross] pr…",
            "url": "https://hackaday.com/2021/11/03/python-ditches-the-gils-and-comes-ashore/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2017/04/python-ide-ask-hackaday-featured1.jpg",
            "publishedAt": "2021-11-03T14:00:56Z",
            "content": "The Python world has been fractured a few times before. The infamous transition from version 2 to version 3 still affects people today, and there could be a new schism in the future. [Sam Gross] prop… [+5724 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Ryan Flowers",
            "title": "Autonomous Drone Dodges Obstacles Without GPS",
            "description": "If you’re [Nick Rehm], you want a drone that can plan its own routes even at low altitudes with unplanned obstacles blocking its way. (Video, embedded below.) And or course, you build it from…",
            "url": "https://hackaday.com/2021/11/03/autonomous-drone-dodges-obstacles-without-gps/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/10/gpsless-drone-featured.jpg",
            "publishedAt": "2021-11-03T11:00:00Z",
            "content": "If you’re [Nick Rehm], you want a drone that can plan its own routes even at low altitudes with unplanned obstacles blocking its way. (Video, embedded below.) And or course, you build it from scratch… [+1327 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Tom Nardi",
            "title": "Liberating the ESP8266 from its Development Board",
            "description": "While the ESP32 is clearly a superior piece of hardware, we think you’ll agree that the ESP8266 is just too useful not to have a dozen or so kicking around the parts bin at any given time. Ch…",
            "url": "https://hackaday.com/2021/11/03/liberating-the-esp8266-from-its-development-board/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/10/esptips_feat.jpg",
            "publishedAt": "2021-11-03T08:00:00Z",
            "content": "While the ESP32 is clearly a superior piece of hardware, we think you’ll agree that the ESP8266 is just too useful not to have a dozen or so kicking around the parts bin at any given time. Cheap, eas… [+1780 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Robin Kearey",
            "title": "Cute Little iMac Clone Runs MacOS on a Tiny Screen",
            "description": "Building a Hackintosh – a non-Apple computer running MacOS – has been a favorite pastime of hackers ever since Apple made the switch from PowerPC to Intel hardware. Though usually …read more",
            "url": "https://hackaday.com/2021/11/02/cute-little-imac-clone-runs-macos-on-a-tiny-screen/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/11/Mini-iMac.jpg",
            "publishedAt": "2021-11-03T05:00:51Z",
            "content": "Building a Hackintosh – a non-Apple computer running MacOS – has been a favorite pastime of hackers ever since Apple made the switch from PowerPC to Intel hardware. Though usually built from commodit… [+1651 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Kristina Panos",
            "title": "Malamud’s General Index: Research Gist, No Slap On the Wrist",
            "description": "Tired of that unsettling feeling you get from looking for paywalled papers on that one site that shall not be named? Yeah, us too. But now there’s an alternative that should feel a little les…",
            "url": "https://hackaday.com/2021/11/02/malamuds-general-index-research-gist-no-slap-on-the-wrist/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/10/general-index-800.jpg",
            "publishedAt": "2021-11-03T02:00:41Z",
            "content": "Tired of that unsettling feeling you get from looking for paywalled papers on that one site that shall not be named? Yeah, us too. But now there’s an alternative that should feel a little less illega… [+1756 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Tom Nardi",
            "title": "Tiny Open Hardware Linux SBC Hides In Plain Sight",
            "description": "There was a time, not quite so long ago, when a computer was a beige box that sat on your desk. Before that, computers were big enough to double as desks, and even farther back, they took up a whol…",
            "url": "https://hackaday.com/2021/11/02/tiny-open-hardware-linux-sbc-hides-in-plain-sight/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/10/wwartprize_feat.jpg",
            "publishedAt": "2021-11-02T23:01:59Z",
            "content": "There was a time, not quite so long ago, when a computer was a beige box that sat on your desk. Before that, computers were big enough to double as desks, and even farther back, they took up a whole … [+2001 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Tom Nardi",
            "title": "Computer Vision Lets You Skip Songs with a Glance",
            "description": "Have you ever wished you could control your home automation devices with nothing more than a withering stare? Well then you’re in luck, as [Norbert Zare] has come up with a clever way of cont…",
            "url": "https://hackaday.com/2021/11/02/computer-vision-lets-you-skip-songs-with-a-glance/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/10/faceskip_feat.jpg",
            "publishedAt": "2021-11-02T20:00:38Z",
            "content": "Have you ever wished you could control your home automation devices with nothing more than a withering stare? Well then you’re in luck, as [Norbert Zare] has come up with a clever way of controlling … [+1447 chars]"
        }
    ]
}
