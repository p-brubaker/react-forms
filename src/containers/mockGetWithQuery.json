{
    "status": "ok",
    "totalResults": 5,
    "articles": [
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Al Williams",
            "title": "RISC-V in… Typescript?",
            "description": "We are accustomed to seeing RISC-V implementations in Verilog or VHDL, but [Low Level JavaScript] has one in TypeScript. Before you dismiss it as a mere emulator, know that the project relies on ga…",
            "url": "https://hackaday.com/2021/10/14/risc-v-in-typescript/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/10/risc.png",
            "publishedAt": "2021-10-15T02:00:15Z",
            "content": "We are accustomed to seeing RISC-V implementations in Verilog or VHDL, but [Low Level JavaScript] has one in TypeScript. Before you dismiss it as a mere emulator, know that the project relies on gate… [+961 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Ryan Flowers",
            "title": "Supply Chain Attack: NPM Library Used By Facebook And Others Was Compromised",
            "description": "Here at Hackaday we love the good kinds of hacks, but now and then we need to bring up a less good kind. Today it was learned that the NPM package ua-parser-js was compromised, and any software usi…",
            "url": "https://hackaday.com/2021/10/22/supply-chain-attack-npm-library-used-by-facebook-and-others-was-compromised/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/10/ua-parser-compromised-featured.jpg",
            "publishedAt": "2021-10-23T02:01:14Z",
            "content": "Here at Hackaday we love the good kinds of hacks, but now and then we need to bring up a less good kind. Today it was learned that the NPM package ua-parser-js was compromised, and any software using… [+2179 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Al Williams",
            "title": "BBC micro:bit Reads Morse Code with MakeCode",
            "description": "We always have mixed feelings about the drag-and-drop programming languages. But we were impressed with [SirDan’s] Morse code decoder built with the graphical MakeCode. Granted, it is reading 5 element …read more",
            "url": "https://hackaday.com/2021/10/06/bbc-microbit-reads-morse-code-with-makecode/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/10/cw.png",
            "publishedAt": "2021-10-07T05:00:39Z",
            "content": "We always have mixed feelings about the drag-and-drop programming languages. But we were impressed with [SirDan’s] Morse code decoder built with the graphical MakeCode. Granted, it is reading 5 eleme… [+1125 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Elliot Williams",
            "title": "Hackers and China",
            "description": "The open source world and Chinese manufacturing have a long relationship. Some fifteen years ago, the big topic was how companies could open-source their hardware designs and not get driven …read more",
            "url": "https://hackaday.com/2021/10/09/hackers-and-china/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2017/08/luban.jpg",
            "publishedAt": "2021-10-09T14:00:11Z",
            "content": "The open source world and Chinese manufacturing have a long relationship. Some fifteen years ago, the big topic was how companies could open-source their hardware designs and not get driven bankrupt … [+1583 chars]"
        },
        {
            "source": {
                "id": null,
                "name": "Hackaday"
            },
            "author": "Elliot Williams",
            "title": "Meet the Winners of The Hackaday Prize Round Four: Redefine Robots",
            "description": "The judges’ ballots are in and we’re proud to present the ten winners of the fourth round of the 2021 Hackaday Prize. We love robots, and it’s obvious that you do too!  The number…",
            "url": "https://hackaday.com/2021/10/07/meet-the-winners-of-the-hackaday-prize-round-four-redefine-robots/",
            "urlToImage": "https://hackaday.com/wp-content/uploads/2021/08/2021-hackaday-prize-challenge-redefine-robots-featured.png",
            "publishedAt": "2021-10-07T16:00:32Z",
            "content": "The judges’ ballots are in and we’re proud to present the ten winners of the fourth round of the 2021 Hackaday Prize. We love robots, and it’s obvious that you do too!  The number and range of projec… [+1976 chars]"
        }
    ]
}
