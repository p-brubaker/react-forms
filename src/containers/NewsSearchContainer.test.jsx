import {
    render,
    screen,
    waitFor,
    waitForElementToBeRemoved,
} from '@testing-library/react';
import React from 'react';
import userEvent from '@testing-library/user-event';
import NewsSearchContainer from './NewsSearchContainer';
import { setupServer } from 'msw/node';
import { rest } from 'msw';

import mockDefaultGetResults from './__mocks__/mockDefaultGetResults.json';
import mockGetWithQuery from './__mocks__/mockGetWithQuery.json';
const URL = 'https://newsapi.org/v2/everything';

describe('Shows news articles', () => {
    const server = setupServer(
        rest.get(URL, (req, res, ctx) => {
            const query = req.url.searchParams;
            const q = query.get('q');

            if (q === ' JavaScript') {
                return res(ctx.status(200), ctx.json(mockGetWithQuery));
            } else {
                return res(ctx.status(200), ctx.json(mockDefaultGetResults));
            }
        })
    );

    beforeAll(() => {
        server.listen({ onUnhandledRequest: 'warn' });
    });

    afterEach(() => server.resetHandlers);

    afterAll(() => {
        server.close();
    });

    it('should display a list of articles from hackaday.com', async () => {
        render(<NewsSearchContainer />);
        screen.getByText('Loading...');

        const ul = await screen.findByRole('list', { name: 'articles' });
        expect(ul).toMatchSnapshot();
    });

    it('should display a list of articles mentioning JavaScript', async () => {
        render(<NewsSearchContainer />);
        screen.getByText('Loading...');

        const queryInput = await screen.findByLabelText('Search by topic');
        userEvent.type(queryInput, 'JavaScript');

        const submitButton = await screen.findByRole('button', {
            name: 'submit',
        });

        userEvent.click(submitButton);

        await waitForElementToBeRemoved(screen.getByText('Loading...'));

        return waitFor(() => {
            const articles = screen.getAllByText('JavaScript', {
                exact: false,
            });

            expect(articles).toHaveLength(1);
        });
    });
});
