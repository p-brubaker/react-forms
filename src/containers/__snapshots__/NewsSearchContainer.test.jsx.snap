// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows news articles should display a list of articles from hackaday.com 1`] = `
<ul
  aria-label="articles"
>
  <li>
    <p>
      Author: 
      Kristina Panos
    </p>
    <p>
      Title: 
      Spooky Winners of the Halloween Hackfest Contest
    </p>
    <p>
      Description: 
      It was a tight race, but the results of the Halloween Hackfest contest are in. We asked you to scare up a terrifying build in one of three categories, and as usual, you didn’t disappoint! The…
    </p>
  </li>
  <li>
    <p>
      Author: 
      Adam Fabio
    </p>
    <p>
      Title: 
      Curious Marc Takes on Sewing Machine Repair
    </p>
    <p>
      Description: 
      Even the most talented engineers can be stymied by simple repair projects. In this case, repairing a broken sewing machine has [CuriousMarc] all tangled up.  [Marc] is probably best known …read more
    </p>
  </li>
  <li>
    <p>
      Author: 
      Chris Wilkinson
    </p>
    <p>
      Title: 
      Weird Al’s Monster Battlestation Is Now Just A Reasonably Fast PC
    </p>
    <p>
      Description: 
      Wanna be hackers? Code crackers? Slackers. If the vintage computing community ever choose an official anthem, count my vote for It’s All About The Pentiums by “Weird Al” Yankovic. More than twenty …
    </p>
  </li>
  <li>
    <p>
      Author: 
      Ryan Flowers
    </p>
    <p>
      Title: 
      Mach 20 In A Wind Tunnel: Yes, It’s Rocket Science
    </p>
    <p>
      Description: 
      Hypersonic speeds are defined by those exceeding Mach 5, and lately there’s been a lot of buzz about unmanned hypersonic vehicles making test flights. Vehicles returning from orbital flight also …read more
    </p>
  </li>
  <li>
    <p>
      Author: 
      Kristina Panos
    </p>
    <p>
      Title: 
      Focus Flower Motivates By Squirting Water In Your Face
    </p>
    <p>
      Description: 
      When you need to get some tasks done and are short on attention, it’s hard to beat a timer. But whenever you do, it feels pretty darn good. The problem is that when you don’t finish in …
    </p>
  </li>
  <li>
    <p>
      Author: 
      Bryan Cockfield
    </p>
    <p>
      Title: 
      Vintage Computers With a Real Turbo
    </p>
    <p>
      Description: 
      In prior centuries, it was common practice to tie the operation of a program to a computer’s clock speed. As computers got faster and faster, the programs tied to that slower clock speed some…
    </p>
  </li>
  <li>
    <p>
      Author: 
      Bryan Cockfield
    </p>
    <p>
      Title: 
      More Software-Defined Radio Projects Using DragonOS
    </p>
    <p>
      Description: 
      DragonOS, a Debian-based Linux distribution specifically packaged for software-defined radio functionality, roared onto the wavelengths during the beginnings of the various pandemic lockdowns last year. Since then [Aaron], the creator …read more
    </p>
  </li>
  <li>
    <p>
      Author: 
      Lewin Day
    </p>
    <p>
      Title: 
      Hacking the Mekamon Robot To Add New Capabilities
    </p>
    <p>
      Description: 
      The Mekamon from Reach Robotics is a neat thing, a robot controlled by a phone app that walks on four legs. [Wes Freeman] decided to hack the platform, giving it a sensor package and enabling some …
    </p>
  </li>
  <li>
    <p>
      Author: 
      Tom Nardi
    </p>
    <p>
      Title: 
      Building a Heavy-Duty Log Splitter, One Piece at a Time
    </p>
    <p>
      Description: 
      With temperatures dropping in the Northern Hemisphere, this is the time of year when many people start processing firewood for the coming winter months. For the city folks, that means chopping a tr…
    </p>
  </li>
  <li>
    <p>
      Author: 
      Adam Zeloof
    </p>
    <p>
      Title: 
      Pringles Can Fluidized Bed Powder Coater Paints Parts Inside And Out
    </p>
    <p>
      Description: 
      Powder coating is a wonderful way to apply a smooth, colored surface to a part, whether for aesthetic reasons or corrosion protection. Traditionally, powder is applied via a air gun that sprays it …
    </p>
  </li>
  <li>
    <p>
      Author: 
      Michael Shaub
    </p>
    <p>
      Title: 
      Keep Calm and Hack On: The Philosophy of Calm Technology
    </p>
    <p>
      Description: 
      So much smart-tech is really kind of dumb. Gadgets intended to simplify our lives turn out to complicate them. It often takes too many “clicks” to accomplish simple tasks, and …read more
    </p>
  </li>
  <li>
    <p>
      Author: 
      Tom Nardi
    </p>
    <p>
      Title: 
      Here Are the Ten Finalists For the Hackaday Prize Wildcard Challenge
    </p>
    <p>
      Description: 
      Each phase of the 2021 Hackaday Prize challenged designers to reimagine traditional solutions within various fields, from robotics to assistive devices. But for the Reactivate Wildcard, the fifth and final …read more
    </p>
  </li>
  <li>
    <p>
      Author: 
      Dave Rowntree
    </p>
    <p>
      Title: 
      Simplify 3D Printer Wiring With CAN Bus
    </p>
    <p>
      Description: 
      [mark] had an interesting idea when looking at all the wiring of a typical 3D printer; Use CAN Bus. There are a lot of wires going to the extruder assembly, and with most designs this thing is flyi…
    </p>
  </li>
  <li>
    <p>
      Author: 
      Matthew Carlson
    </p>
    <p>
      Title: 
      Python Ditches the GILs and Comes Ashore
    </p>
    <p>
      Description: 
      The Python world has been fractured a few times before. The infamous transition from version 2 to version 3 still affects people today, and there could be a new schism in the future. [Sam Gross] pr…
    </p>
  </li>
  <li>
    <p>
      Author: 
      Ryan Flowers
    </p>
    <p>
      Title: 
      Autonomous Drone Dodges Obstacles Without GPS
    </p>
    <p>
      Description: 
      If you’re [Nick Rehm], you want a drone that can plan its own routes even at low altitudes with unplanned obstacles blocking its way. (Video, embedded below.) And or course, you build it from…
    </p>
  </li>
  <li>
    <p>
      Author: 
      Tom Nardi
    </p>
    <p>
      Title: 
      Liberating the ESP8266 from its Development Board
    </p>
    <p>
      Description: 
      While the ESP32 is clearly a superior piece of hardware, we think you’ll agree that the ESP8266 is just too useful not to have a dozen or so kicking around the parts bin at any given time. Ch…
    </p>
  </li>
  <li>
    <p>
      Author: 
      Robin Kearey
    </p>
    <p>
      Title: 
      Cute Little iMac Clone Runs MacOS on a Tiny Screen
    </p>
    <p>
      Description: 
      Building a Hackintosh – a non-Apple computer running MacOS – has been a favorite pastime of hackers ever since Apple made the switch from PowerPC to Intel hardware. Though usually …read more
    </p>
  </li>
  <li>
    <p>
      Author: 
      Kristina Panos
    </p>
    <p>
      Title: 
      Malamud’s General Index: Research Gist, No Slap On the Wrist
    </p>
    <p>
      Description: 
      Tired of that unsettling feeling you get from looking for paywalled papers on that one site that shall not be named? Yeah, us too. But now there’s an alternative that should feel a little les…
    </p>
  </li>
  <li>
    <p>
      Author: 
      Tom Nardi
    </p>
    <p>
      Title: 
      Tiny Open Hardware Linux SBC Hides In Plain Sight
    </p>
    <p>
      Description: 
      There was a time, not quite so long ago, when a computer was a beige box that sat on your desk. Before that, computers were big enough to double as desks, and even farther back, they took up a whol…
    </p>
  </li>
  <li>
    <p>
      Author: 
      Tom Nardi
    </p>
    <p>
      Title: 
      Computer Vision Lets You Skip Songs with a Glance
    </p>
    <p>
      Description: 
      Have you ever wished you could control your home automation devices with nothing more than a withering stare? Well then you’re in luck, as [Norbert Zare] has come up with a clever way of cont…
    </p>
  </li>
</ul>
`;
